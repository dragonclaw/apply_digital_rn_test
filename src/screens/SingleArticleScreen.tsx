import {StyleSheet, Text, View} from 'react-native';
import React from 'react';

const SingleArticleScreen = () => {
  return (
    <View>
      <Text>SingleArticleScreen</Text>
    </View>
  );
};

export default SingleArticleScreen;

const styles = StyleSheet.create({});
