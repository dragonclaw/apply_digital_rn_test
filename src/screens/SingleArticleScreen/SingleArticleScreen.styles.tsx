import styled from 'styled-components';
import {View} from 'react-native';

export const WebViewContainer = styled(View)`
  flex: 1;
`;
