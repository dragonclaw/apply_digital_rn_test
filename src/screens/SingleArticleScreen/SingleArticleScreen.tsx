import React from 'react';
import {useRoute, RouteProp} from '@react-navigation/native';
import {WebView} from 'react-native-webview';
import {WebViewContainer} from './SingleArticleScreen.styles';
import {Text} from '@rneui/themed';

type RootStackParamList = {
  SingleArticleScreen: {article: any};
};

type SingleArticleScreenRouteProp = RouteProp<
  RootStackParamList,
  'SingleArticleScreen'
>;

const SingleArticleScreen = () => {
  const route = useRoute<SingleArticleScreenRouteProp>();
  const {article} = route.params;

  return (
    <WebViewContainer>
      {article.url ? (
        <WebView
          source={{
            uri: article.url ?? '',
          }}
        />
      ) : (
        //TODO: Add a better error handling component
        <>
          <Text> No URL on this story</Text>
        </>
      )}
    </WebViewContainer>
  );
};

export default SingleArticleScreen;
