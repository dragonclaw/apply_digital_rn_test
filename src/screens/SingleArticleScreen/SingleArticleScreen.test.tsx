import React from 'react';
import {render} from '@testing-library/react-native';
import {useRoute} from '@react-navigation/native';
import SingleArticleScreen from './SingleArticleScreen'; // Adjust the import path as necessary

// Mock the useRoute hook
jest.mock('@react-navigation/native', () => ({
  useRoute: jest.fn(),
}));

jest.mock('react-native-webview', () => {
  const {View} = require('react-native');
  return {
    WebView: ({source}: {source: {uri: string}}) => (
      <View testID="webview">{source.uri}</View>
    ),
  };
});

describe('SingleArticleScreen', () => {
  it('matches the snapshot when URL is provided', () => {
    // Mock the route params
    (useRoute as jest.Mock).mockReturnValue({
      params: {url: 'https://google.com'},
    });

    const {toJSON} = render(<SingleArticleScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches the snapshot when URL is not provided', () => {
    // Mock the route params
    (useRoute as jest.Mock).mockReturnValue({
      params: {url: ''},
    });

    const {toJSON} = render(<SingleArticleScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
