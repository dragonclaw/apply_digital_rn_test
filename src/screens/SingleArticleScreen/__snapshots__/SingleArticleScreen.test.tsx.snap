// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleArticleScreen matches the snapshot when URL is not provided 1`] = `
<View
  className="sc-braxZu ccBOXj"
>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#242424",
      }
    }
  >
     No URL on this story
  </Text>
</View>
`;

exports[`SingleArticleScreen matches the snapshot when URL is provided 1`] = `
<View
  className="sc-braxZu ccBOXj"
>
  <View
    testID="webview"
  >
    https://example.com
  </View>
</View>
`;
